Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Beckn/ACL/OnUpdate.hs

```diff
@@ -69,18 +72,26 @@ data OnUpdateBuildReq
       }
 
 buildOnUpdateMessage ::
-  (EsqDBFlow m r, EncFlow m r) =>
+  (EsqDBFlow m r, EncFlow m r, EsqDBReplicaFlow m r) =>
   OnUpdateBuildReq ->
   m OnUpdate.OnUpdateMessage
 buildOnUpdateMessage RideAssignedBuildReq {..} = do
   mobileNumber <- SP.getPersonNumber driver >>= fromMaybeM (InternalError "Driver mobile number is not present.")
   name <- SP.getPersonFullName driver >>= fromMaybeM (PersonFieldNotPresent "firstName")
+  inProgressRide <- Ride.getInProgressByDriverId driver.id
```

Comment:
why is this query required ?

Reply:
To check the driver.id has one ride in progress state and one ride in new state

Reply:
but do we need to check this, i.e. if there is no inprogress but newRide, then the `findByDriverIdTripEndLatLonIfRideStatusNew` query will return Nothing right..? so we only need two queries to get this, in current way we will do 2 queries always and in some cases 3.

Reply:
I think because of lazy evaluation it will not call the query so I have written the code like this

Reply:
m asking something else, where is lazy evaluation coming here ?